mobile/src/reducers/auth/auth.reducer.ts

```diff
@@ -0,0 +1,36 @@
+import { Reducer } from 'redux';
+import { combineEpics } from 'redux-observable';
+import { pressLogInEpic$, pressLogOutEpic$, listenerAuthStateEpic$ } from './auth.epics';
+import { AuthActions } from './auth.action';
+import { AuthenticationState } from '../../auth/authentication-state';
+
+export const authEpics$ = combineEpics(
+    pressLogInEpic$ as any,
+    pressLogOutEpic$ as any,
+    listenerAuthStateEpic$ as any
+);
+
+export interface AuthState {
+    isAuthenticated: boolean | null;
```

Comment:
Why do we need `null` state?

Reply:
It is for splash screen. We need 3 states: 
first one - user is authenticated (isAuthenticated: true)
second one - user is not authenticated (isAuthenticated: false)
third one - app is launching and we have not receive authentication state yet  (isAuthenticated: null).

Reply:
That's actually wrong. `isAuthenticated` should respond only to whether the user is authenticated or not, it shouldn't know about other states such as app is loaded. You should use the other state to track whether app is loaded

Reply:
Quick solution, use a substate:
```javascript
interface AuthInfo {
   isAuthenticated: boolean
}

interface AuthState {
  authInfo: AuthInfo | null;
}

public render() {
     if (!this.props.authentication.authInfo) {
          return <SplashScreen />
     }

     } else if (this.props.authentication.authInfo.isAuthenticated ) {
       return <RootNavigator />;
     } else {
       return <WelcomeScreen />;
     }
   }	   
}
```

Reply:
well, what you've done here is traded `bool` for `AuthInfo` :) Since it's easier to read, I do not mind the change.